xmtp/src/persistence/mod.rs

```diff
@@ -1,8 +1,9 @@
 pub mod in_memory_persistence;
 
 pub trait Persistence {
-    fn write(&mut self, key: &str, value: &[u8]) -> Result<(), String>;
-    fn read(&self, key: &str) -> Result<Option<Vec<u8>>, String>;
+    type Error;
```

Comment:
I can get behind using Associated Types for Errors. Seems like in most cases the Errors could be standardized at the trait level which would simplify handling, thought I imagine we expect there to be many error types specific to the Trait Impl that need handling?